src/DIRAC/WorkloadManagementSystem/DB/VirtualMachineDB.py

```diff
@@ -0,0 +1,1435 @@
+
+  def insertInstance(self, uniqueID, imageName, instanceName, endpoint, runningPodName):
+    """
+    Check Status of a given image
+    Will insert a new Instance in the DB
+    returns:
+      S_OK( InstanceID ) if new Instance is properly inserted
+      S_ERROR(ErrorMessage) otherwise
+    """
+    imageStatus = self.checkImageStatus(imageName)
+    if not imageStatus['OK']:
+      return imageStatus
+
+    return self.__insertInstance(uniqueID, imageName, instanceName, endpoint, runningPodName)
+
+  def setInstanceUniqueID(self, instanceID, uniqueID):
+    """
+    Assign a uniqueID to an instance
+    """
+    result = self.__getInstanceID(uniqueID)
+    if result['OK']:
+      return S_ERROR('UniqueID is not unique: %s' % uniqueID)
+
+    result = self._escapeString(uniqueID)
+    if not result['OK']:
+      return result
+    uniqueID = result['Value']
+
+    try:
+      instanceID = int(instanceID)
+    except ValueError:
+      # except Exception, e:
+      # FIXME: do we really want to raise an Exception ?
+      # raise e
+      return S_ERROR("instanceID has to be a number")
+
+    tableName, _validStates, idName = self.__getTypeTuple('Instance')
+
+    sqlUpdate = "UPDATE `%s` SET UniqueID = %s WHERE %s = %d" % (tableName, uniqueID, idName, instanceID)
+    return self._update(sqlUpdate)
+
+  def getInstanceParameter(self, pName, instanceID):
+    """ Get the instance parameter pName for the given instanceID
+    :param pName: parameter name
+    :param instance ID: instance ID
+    :return: S_OK/S_ERROR, parameter value
+    """
+
+    tableName, _validStates, idName = self.__getTypeTuple('Instance')
+
+    if pName not in VirtualMachineDB.tablesDesc['vm_Instances']['Fields']:
+      return S_ERROR('Invalid Instance parameter %s' % pName)
+
+    sqlQuery = "SELECT %s FROM `%s` WHERE %s = %s" % (pName, tableName, idName, instanceID)
+    result = self._query(sqlQuery)
+
+    if not result['OK']:
+      return result
+    value = result['Value'][0][0]
+
+    return S_OK(value)
+
+  def getUniqueID(self, instanceID):
+    """
+    For a given dirac instanceID get the corresponding cloud endpoint uniqueID
+    """
+    return self.getInstanceParameter('UniqueID', instanceID)
+
+  def getUniqueIDByName(self, instanceName):
+    """ Get the cloud provider unique ID corresponding to the DIRAC unique name
+    :param name: VM name
+    :return: S_OK/S_ERROR, cloud unique ID as value
+    """
+    tableName, _validStates, idName = self.__getTypeTuple('Instance')
+
+    sqlQuery = "SELECT UniqueID FROM `%s` WHERE Name = '%s'" % (tableName, instanceName)
+    result = self._query(sqlQuery)
+    if not result['OK']:
+      return result
+    if not result['Value']:
+      return S_ERROR('Instance not found')
+    uniqueID = result['Value'][0][0]
+
+    return S_OK(uniqueID)
+
+  def getInstanceID(self, uniqueID):
+    """
+    Public interface for  __getInstanceID
+    """
+    return self.__getInstanceID(uniqueID)
```

Comment:
...then what is the point of having `__getInstanceID`?

Reply:
Gone, merged private function into public function and updated references.